src/auto_archiver/archivers/archiver.py

```diff
@@ -59,11 +67,28 @@ def download_from_url(self, url: str, to_filename: str = None, verbose=True) ->
         headers = {
             'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36'
         }
-        d = requests.get(url, headers=headers)
-        assert d.status_code == 200, f"got response code {d.status_code} for {url=}"
-        with open(to_filename, 'wb') as f:
-            f.write(d.content)
-        return to_filename
+        try:
+            d = requests.get(url, stream=True, headers=headers)
+            d.raise_for_status()
+
+            # Peek at the first 256 bytes
+            first_256 = d.raw.read(256)
```

Comment:
is this a standard size for extension guessing?

Reply:
This was a typo, it should be 261, all that's [required to detect](https://github.com/h2non/filetype.py?tab=readme-ov-file#features). But yes, that's all that's needed to guess filetype. No point loading more than that (and wasting memory) to determine the filetype!

Reply:
interesting!
I haven't tested but did you check if mimetypes can achieve the same thing with file chunks? seeing this method https://docs.python.org/3/library/mimetypes.html#mimetypes.MimeTypes.readfp may be worth quickly testing to avoid adding a new dependency